Flaky test TestJdbcClient.testAlterColumns

Test failed with the following error: 
```
[ERROR] Tests run: 64, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 45.693 s <<< FAILURE! - in TestSuite
2021-05-12T14:32:29.5246009Z [ERROR] testAlterColumns(com.facebook.presto.plugin.jdbc.TestJdbcClient)  Time elapsed: 0.003 s  <<< FAILURE!
2021-05-12T14:32:29.5247881Z com.facebook.presto.spi.PrestoException: 
2021-05-12T14:32:29.5250868Z Timeout trying to lock table "SYS"; SQL statement:
2021-05-12T14:32:29.5262610Z ALTER TABLE "TEST724489858213"."SCHEMA_FOR_CREATE_TABLE_TESTS"."E84FE40F-51E3-45F6-A60C-DE6E3B5A5BB0" ADD "COLUMNB" double precision [50200-199]
2021-05-12T14:32:29.5267614Z 	at com.facebook.presto.plugin.jdbc.BaseJdbcClient.addColumn(BaseJdbcClient.java:492)
2021-05-12T14:32:29.5283654Z 	at com.facebook.presto.plugin.jdbc.TestJdbcClient.testAlterColumns(TestJdbcClient.java:171)
2021-05-12T14:32:29.5286986Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2021-05-12T14:32:29.5292211Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2021-05-12T14:32:29.5297845Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-05-12T14:32:29.5304390Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2021-05-12T14:32:29.5310427Z 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
2021-05-12T14:32:29.5313458Z 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
2021-05-12T14:32:29.5317408Z 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
2021-05-12T14:32:29.5321457Z 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
2021-05-12T14:32:29.5326904Z 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
2021-05-12T14:32:29.5330927Z 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
2021-05-12T14:32:29.5336364Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2021-05-12T14:32:29.5341147Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2021-05-12T14:32:29.5342943Z 	at java.lang.Thread.run(Thread.java:748)
2021-05-12T14:32:29.5348864Z Caused by: org.h2.jdbc.JdbcSQLTimeoutException: Timeout trying to lock table "SYS"; SQL statement:
2021-05-12T14:32:29.5359189Z ALTER TABLE "TEST724489858213"."SCHEMA_FOR_CREATE_TABLE_TESTS"."E84FE40F-51E3-45F6-A60C-DE6E3B5A5BB0" ADD "COLUMNB" double precision [50200-199]
2021-05-12T14:32:29.5362945Z 	at org.h2.message.DbException.getJdbcSQLException(DbException.java:508)
2021-05-12T14:32:29.5367361Z 	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
2021-05-12T14:32:29.5370297Z 	at org.h2.message.DbException.get(DbException.java:205)
2021-05-12T14:32:29.5373763Z 	at org.h2.message.DbException.get(DbException.java:181)
2021-05-12T14:32:29.5377236Z 	at org.h2.mvstore.db.MVTable.doLock1(MVTable.java:221)
2021-05-12T14:32:29.5380440Z 	at org.h2.mvstore.db.MVTable.lock(MVTable.java:165)
2021-05-12T14:32:29.5384433Z 	at org.h2.engine.Database.lockMeta(Database.java:1079)
2021-05-12T14:32:29.5387200Z 	at org.h2.schema.Schema.createTable(Schema.java:691)
2021-05-12T14:32:29.5394582Z 	at org.h2.command.ddl.AlterTableAlterColumn.cloneTableStructure(AlterTableAlterColumn.java:399)
2021-05-12T14:32:29.5404412Z 	at org.h2.command.ddl.AlterTableAlterColumn.copyData(AlterTableAlterColumn.java:286)
2021-05-12T14:32:29.5412274Z 	at org.h2.command.ddl.AlterTableAlterColumn.update(AlterTableAlterColumn.java:201)
2021-05-12T14:32:29.5419349Z 	at org.h2.command.CommandContainer.update(CommandContainer.java:133)
2021-05-12T14:32:29.5425099Z 	at org.h2.command.Command.executeUpdate(Command.java:267)
2021-05-12T14:32:29.5432533Z 	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:233)
2021-05-12T14:32:29.5440258Z 	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:205)
2021-05-12T14:32:29.5447864Z 	at com.facebook.presto.plugin.jdbc.BaseJdbcClient.execute(BaseJdbcClient.java:710)
2021-05-12T14:32:29.5457610Z 	at com.facebook.presto.plugin.jdbc.BaseJdbcClient.addColumn(BaseJdbcClient.java:489)
2021-05-12T14:32:29.5460798Z 	... 14 more
2021-05-12T14:32:29.5462089Z 
2021-05-12T14:32:29.8994819Z [INFO] 
2021-05-12T14:32:29.8995634Z [INFO] Results:
2021-05-12T14:32:29.8996140Z [INFO] 
2021-05-12T14:32:29.8996637Z [ERROR] Failures: 
2021-05-12T14:32:29.9036025Z [ERROR]   TestJdbcClient.testAlterColumns:171 » Presto Timeout trying to lock table "SYS...
```

--- a/presto-base-jdbc/src/test/java/com/facebook/presto/plugin/jdbc/TestJdbcClient.java
+++ b/presto-base-jdbc/src/test/java/com/facebook/presto/plugin/jdbc/TestJdbcClient.java
@@ -151,8 +151,7 @@ public class TestJdbcClient
         }
     }
 
-    // disabled due to https://github.com/prestodb/presto/issues/16081
-    @Test(enabled = false)
+    @Test
     public void testAlterColumns()
     {
         String tableName = randomUUID().toString().toUpperCase(ENGLISH);

--- a/presto-base-jdbc/src/test/java/com/facebook/presto/plugin/jdbc/TestingDatabase.java
+++ b/presto-base-jdbc/src/test/java/com/facebook/presto/plugin/jdbc/TestingDatabase.java
@@ -18,6 +18,7 @@ import com.facebook.presto.spi.ConnectorSession;
 import com.facebook.presto.spi.ConnectorSplitSource;
 import com.facebook.presto.spi.SchemaTableName;
 import com.google.common.collect.ImmutableMap;
+import io.airlift.units.Duration;
 import org.h2.Driver;
 
 import java.sql.Connection;
@@ -34,10 +35,12 @@ import static com.facebook.presto.spi.connector.NotPartitionedPartitionHandle.NO
 import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.collect.Iterables.getOnlyElement;
+import static java.util.concurrent.TimeUnit.SECONDS;
 
 final class TestingDatabase
         implements AutoCloseable
 {
+    public static final Duration LOCK_TIMEOUT = new Duration(30, SECONDS);
     public static final String CONNECTOR_ID = "test";
     private static final ConnectorSession session = testSessionBuilder().build().toConnectorSession();
 
@@ -47,7 +50,7 @@ final class TestingDatabase
     public TestingDatabase()
             throws SQLException
     {
-        String connectionUrl = "jdbc:h2:mem:test" + System.nanoTime() + "_" + ThreadLocalRandom.current().nextInt();
+        String connectionUrl = "jdbc:h2:mem:test" + System.nanoTime() + "_" + ThreadLocalRandom.current().nextInt() + ";LOCK_TIMEOUT=" + LOCK_TIMEOUT.toMillis();
         jdbcClient = new BaseJdbcClient(
                 new JdbcConnectorId(CONNECTOR_ID),
                 new BaseJdbcConfig(),

